justfile

```diff
@@ -67,14 +67,31 @@ check-wasm:
 [no-exit-message]
 typos:
   #!/usr/bin/env bash
-  >&2 echo '💡 Valid new words can be added to `_typos.toml`'
-  typos
+  set -eo pipefail
+
+  git_ls_files="$(git ls-files)"
+  git_ls_nonbinary_files="$(echo "$git_ls_files" | xargs file --mime | grep -v "; charset=binary" | cut -d: -f1)"
+
+  if ! echo "$git_ls_nonbinary_files" | parallel typos {} ; then
+    >&2 echo "Typos found: Valid new words can be added to '_typos.toml'"
+    # TODO: not enforcing anything right, just being annoying in the CLI
+    # return 1
+  fi
+
 
 [no-exit-message]
 typos-fix-all:
   #!/usr/bin/env bash
-  >&2 echo '💡 Valid new words can be added to `_typos.toml`'
-  typos --write-changes
+  set -eo pipefail
+
+  git_ls_files="$(git ls-files)"
+  git_ls_nonbinary_files="$(echo "$git_ls_files" | xargs file --mime | grep -v "; charset=binary" | cut -d: -f1)"
+
+  if ! echo "$git_ls_nonbinary_files" | parallel typos -w {} ; then
+    >&2 echo "Typos found: Valid new words can be added to '_typos.toml'"
+    # TODO: not enforcing anything right, just being annoying in the CLI
+    # return 1
+  fi
```

Comment:
Seems copy-pasted, could it be a function?

Reply:
Will maybe have to move it to a script. But it's actually too slow for my liking, and I have a PR upstream to make it better anyway, so it might this PR might need to change a lot anyway. https://github.com/crate-ci/typos/pull/708